cmd/common/registry.go

```diff
@@ -0,0 +1,340 @@
+					case utils.OutputModeJSON:
+						fmt.Fprint(os.Stdout, string(res))
+					}
+				}()
+			}
+
+			// Finally, hand over to runtime
+			return runner.RunGadget(runtimeParams, operatorsParamCollection, operatorsPerGadgetParamCollection, params)
+		},
+	}
+
+	if gadget.Type() != gadgets.TypeOneShot {
+		// Add timeout
+		cmd.PersistentFlags().IntVarP(&timeout, "timeout", "t", 0, "Number of seconds that the gadget will run for, 0 to disable")
+	}
+
+	cmd.PersistentFlags().BoolVarP(
+		&verbose,
+		"verbose", "v",
+		false,
+		"Print debug information",
+	)
+
+	outputFormats.Append(gadgets.OutputFormats{
+		"json": {
+			Name:        "JSON",
+			Description: "The output of the gadget is returned as raw JSON",
+			Transform:   nil,
+		},
+	})
+	defaultOutputFormat = "json"
+
+	// Add output flags
+	if columns != nil {
+		outputFormats.Append(gadgets.OutputFormats{
+			"columns": {
+				Name:        "Columns",
+				Description: "The output of the gadget is formatted in human readable columns",
+				Transform:   nil,
+			},
+		})
+		defaultOutputFormat = "columns"
+
+		cmd.PersistentFlags().StringSliceVarP(
+			&showColumns,
+			"columns", "C",
+			columns.GetDefaultColumns(),
+			"Columns to output",
+		)
+		cmd.PersistentFlags().StringSliceVarP(
+			&filters,
+			"filter", "F",
+			[]string{},
+			"Filter rules",
+		)
+
+		// Add params matching the gadget type
+		params.AddParams(gadgets.GadgetParams(gadget, columns))
+	}
+
+	// Add alternative output formats available in the gadgets
+	if outputFormatInterface, ok := gadget.(gadgets.GadgetOutputFormats); ok {
+		formats, defaultFormat := outputFormatInterface.OutputFormats()
+		outputFormats.Append(formats)
+		defaultOutputFormat = defaultFormat
+	}
+
+	var supportedOutputFormats []string
+	var outputFomatsHelp []string
+
+	for ofKey, of := range outputFormats {
+		supportedOutputFormats = append(supportedOutputFormats, ofKey)
+		desc := fmt.Sprintf("%s (%s)", of.Name, ofKey)
+		if of.Description != "" {
+			desc += fmt.Sprintf("\n  %s", of.Description)
+		}
+		outputFomatsHelp = append(outputFomatsHelp, desc)
+	}
+	sort.Strings(outputFomatsHelp)
+	outputFomatsHelp = append([]string{fmt.Sprintf("Output format (%s).", strings.Join(supportedOutputFormats, ", ")), ""}, outputFomatsHelp...)
+
+	cmd.PersistentFlags().StringVarP(
+		&outputMode,
+		"output",
+		"o",
+		defaultOutputFormat,
+		strings.Join(outputFomatsHelp, "\n")+"\n\n",
+	)
+
+	// Add gadget flags
+	addFlags(cmd, params)
+
+	// Add per-gadget operator flags
+	for _, operatorParams := range operatorsPerGadgetParamCollection {
+		addFlags(cmd, operatorParams)
+	}
+	return cmd
+}
+
+func addFlags(cmd *cobra.Command, params params.Params) {
+	for _, p := range params {
+		desc := p.Description
+
+		if p.PossibleValues != nil {
+			desc += " [" + strings.Join(p.PossibleValues, ", ") + "]"
+		}
+
+		flag := cmd.PersistentFlags().VarPF(p, p.Key, p.Alias, desc)
+		if p.IsBoolFlag() {
+			flag.NoOptDefVal = "true"
+		}
```

Comment:
It took me a while to understand what it was for. Perhaps the following comment helps?

```suggestion
		// Allow passing a boolean flag as --foo instead of having to use --foo=true
		if p.IsBoolFlag() {
			flag.NoOptDefVal = "true"
		}
```